docs/care/CEP/Draft/0010-labs.md

```diff
@@ -0,0 +1,370 @@
+class SampleType(BaseModel):
+    """
+    Model to store the type of sample that can be collected for an investigation. For example, Blood, Urine, etc.
+
+    The data in this model is expected to be static and should be populated via fixtures or migrations.
+    """
+
+    name = models.CharField(max_length=100, unique=True)
+    description = models.TextField(blank=True, null=True)
+
+    def __str__(self):
+        return self.name
+
+
+class InvestigationGroup(BaseModel):
+    """
+    Model to store the group of investigations. For example, Hematology, Biochemistry, etc.
+
+    The data in this model is expected to be static and should be populated via fixtures or migrations.
+    """
+
+    name = models.CharField(max_length=100, unique=True)
+    description = models.TextField(blank=True, null=True)
+
+    def __str__(self):
+        return self.name
+
+
+class Investigation(BaseModel):
+    """
+    Model to store the investigations that can be ordered for a patient. For example, Hemoglobin Count, Blood Sugar Level, etc. This model also stores the sample type required for the investigation, the unit of measurement, the result type, the default min and max values for the result (if the result type is NUMERIC), and the choices for the result (if the result type is CHOICE).
+
+    The data in this model is expected to be static and should be populated via fixtures or migrations.
+    """
+
+    name = models.CharField(max_length=200, unique=True)
+    group = models.ForeignKey(InvestigationGroup, on_delete=models.PROTECT)
+    sample_type = models.ForeignKey(SampleType, on_delete=models.PROTECT)
+    unit = models.CharField(max_length=50, blank=True, null=True)
+    result_type = models.CharField(
+        max_length=20,
+        choices=[
+            ("NUMERIC", "Numeric"),
+            ("TEXT", "Text"),
+            ("CHOICE", "Choice"),
+        ],
+    )
+    default_min_value = models.FloatField(blank=True, null=True)
+    default_max_value = models.FloatField(blank=True, null=True)
+    choices = models.TextField(blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.name} ({self.group.name})"
+
+
+class LabOrder(BaseModel):
+    """
+    Model to store the lab orders ordered by the doctor for the patient. The lab order can be for one or more investigations. The lab order can be for a single consultation or can span across multiple consultations. The lab order can be a one-time order or a repeating order.
+    """
+
+    # if spanning across multiple consultation is linear, consultations in b/w are not skipped
+    starting_consultation = models.ForeignKey(
+        PatientConsultation, on_delete=models.PROTECT, related_name="started_lab_orders"
+    )
+    ending_consultation = models.ForeignKey(
+        PatientConsultation,
+        on_delete=models.PROTECT,
+        related_name="ended_lab_orders",
+        null=True,
+        blank=True,
+    )
+    # else
+    # one to many relation with LabOrderConsultation
```

Comment:
Why do we need relation to multiple consultations?

Reply:
This actually was a requirement mentioned under `Requirements`, `8. Integration with Existing Modules`.

Reply:
Okay. It is better to do 1 primary consultation and others can be a one-to-many relation. `ending_consultation` is bad design, you should never expect a scenario where all consultations between starting and ending will be for the same patient.